pulsar-client-cpp/lib/OpSendMsg.h

```diff
@@ -25,9 +25,26 @@
 
 #include "TimeUtils.h"
 #include "MessageImpl.h"
+#include "ChunkMessageIdImpl.h"
 
 namespace pulsar {
 
+struct ProducerChunkedMessageCtx {
+   public:
+    ProducerChunkedMessageCtx() = default;
+
+    ProducerChunkedMessageCtx(MessageIdImplPtr firstChunkMessageId, MessageIdImplPtr lastChunkMessageId)
+        : firstChunkMessageIdImplPtr_(firstChunkMessageId), lastChunkMessageIdImplPtr_(lastChunkMessageId) {}
+
+    MessageIdImplPtr getChunkMessageId() {
+        return std::make_shared<ChunkMessageIdImpl>(*firstChunkMessageIdImplPtr_, *lastChunkMessageIdImplPtr_)
+            ->getLastChunkMessageIdImpl();
+    }
+
+    MessageIdImplPtr firstChunkMessageIdImplPtr_;
+    MessageIdImplPtr lastChunkMessageIdImplPtr_;
```

Comment:
Make them private. The reason `OpSendMsg`'s fields are not private is historical. Because some fields are accessed directly. I'll add some refactors in future.